src/views/recordTable.component.tsx

```diff
@@ -51,15 +55,30 @@ const RecordTable = React.memo(
 
         for (let i = 0; i < keys.length; i++) {
           if (!accessors.has(keys[i])) {
+            const channelInfo = channels?.find(
+              (channel) => channel.systemName === keys[i]
+            );
```

Comment:
So just to confirm that I have this all in my head correctly:

1. useChannels fetches all the info and metadata about every possible channel that could theoretically be returned to the frontend
2. Lines 58-60 are about matching up a column to a possible channel (with a set of metadata) describing it. Records maintain a list of channel names so we take each channel name and match it with its metadata
3. The matching channel info is added to the column object
4. Once all columns are constructed correctly, they are stored and passed to the Table component

One thing I do want to ask is (assuming I have all this right) is where are we getting the info about the channels in step 1? Will there be a new API endpoint for this?

Reply:
We don't know _exactly_ where we'll be getting to channel info from (either via the og-api or via an external one) but it's definitely not going to be via the records endpoint. E.g. there's going to be a channel selector where the user can select any of the potential channels.

The exact way of columns accessing the data channel metadata I'm still kind of unsure of (see the later Redux PR) as there's two kinds of column info - the "static" info that comes from the server and then the "dynamic" info like the user's word wrap setting for that column. Either way - this way works well enough for this PR and more accurately reflects what the final workflow will be (aka fetching channels separately from records) and doesn't overhaul the existing code too much.